extension/goext/context.go

```diff
@@ -0,0 +1,258 @@
+package goext
+
+import (
+	"context"
+	"net/http"
+)
+
+// Context represents a context of a handler
+type Context interface {
+	GetID() (string, bool)
+	SetID(id string) Context
+
+	GetSchema() (ISchema, bool)
+	SetSchema(schema ISchema) Context
+
+	GetSchemaID() (string, bool)
+	SetSchemaID(schemaID string) Context
+
+	SetISchema(schema ISchema) Context
+	GetISchema() (ISchema, bool)
+	DeleteISchema()
+
+	GetResource() (Resource, bool)
+	SetResource(resource Resource) Context
+	DeleteResource()
+
+	SetTransaction(tx ITransaction) Context
+	GetTransaction() (ITransaction, bool)
+	DeleteTransaction()
+
+	SetContext(context context.Context) Context
+	GetContext() (context.Context, bool)
+	DeleteContext()
+
+	GetIsTopLevelHandler() (bool, bool)
+	SetTopLevelHandler(isTopLevelHandler bool) Context
+	DeleteIsTopLevelHandler()
+
+	GetHttpRequest() (*http.Request, bool)
+	SetHttpRequest(req *http.Request) Context
+
+	GetTenantID() (string, bool)
+	SetTenantID(tenantID string) Context
+
+	GetResponse() (interface{}, bool)
+	SetResponse(response interface{}) Context
+
+	SetInput(input map[string]interface{}) Context
+	GetInput() (map[string]interface{}, bool)
+
+	Clone() Context
+}
+
+type GohanContext map[string]interface{}
+
+func (c GohanContext) SetInput(input map[string]interface{}) Context {
+	c["input"] = input
+	return c
+}
+
+func (c GohanContext) GetInput() (map[string]interface{}, bool) {
+	input, ok := c["input"]
+	if !ok {
+		return nil, ok
+	}
+	return input.(map[string]interface{}), ok
```

Comment:
All getters may panic on type assertion. Are you sure, we don't need special handling, instead (similar to httpRequest)?

Reply:
special handling in httpRequest came from fact that nil isn't convertable to *http.Request. That's true for maps too. I'll add special handling.